src/pybind/mgr/balancer/module.py

```diff
@@ -1255,13 +1258,22 @@ def execute(self, plan):
                 'pgid': pgid,
             }), 'foo')
             commands.append(result)
-
+        
+        # FIXME loop can be optimized
         for item in incdump.get('new_pg_upmap_items', []):
             self.log.info('ceph osd pg-upmap-items %s mappings %s', item['pgid'],
                           item['mappings'])
             osdlist = []
+            which_osds = []
+            affected_pgs = [] 
             for m in item['mappings']:
                 osdlist += [m['from'], m['to']]
+                if m['from'] not in which_osds:
+                    which_osds += [m['from']]
+                temp_list = item['pgid'].split('.')
+                pg_obj = self.remote('progress', 'construct_pgid', pool_id=int(temp_list[0]), ps=int(temp_list[1]))
+                affected_pgs.append(pg_obj)
```

Comment:
`affected_pgs` is defined in this lexical scope.
```
2019-08-08T15:06:12.169 INFO:tasks.ceph.mgr.x.mira088.stderr:2019-08-08T15:06:12.049+0000 7f69075f1700 -1 Traceback (most recent call last):
2019-08-08T15:06:12.169 INFO:tasks.ceph.mgr.x.mira088.stderr:  File "/usr/share/ceph/mgr/balancer/module.py", line 626, in serve
2019-08-08T15:06:12.169 INFO:tasks.ceph.mgr.x.mira088.stderr:    self.execute(plan)
2019-08-08T15:06:12.169 INFO:tasks.ceph.mgr.x.mira088.stderr:  File "/usr/share/ceph/mgr/balancer/module.py", line 1297, in execute
2019-08-08T15:06:12.169 INFO:tasks.ceph.mgr.x.mira088.stderr:    which_pgs=affected_pgs,
2019-08-08T15:06:12.169 INFO:tasks.ceph.mgr.x.mira088.stderr:UnboundLocalError: local variable 'affected_pgs' referenced before assignment
```

see http://pulpito.ceph.com/kchai-2019-08-08_11:54:56-rados-wip-kefu-testing-2019-08-08-1633-distro-basic-mira/4198445/